### DISCUSSION TITLE
KTC Default Tool at Startup

### DISCUSSION THREAD

**Discussion by ClaudeH68** [TypQxQ/KTC#47] [Configuring]
Hi Andrei, first very good job for your KTC. Also, good idea for the custom parameter that way it's easy to expand for the need of the project.  
What I need to know from you, if there is a easy way to default at the startup the tool at TOOL_NONE instead of TOOL_UNKNOW.  I have switch on my heads, so I can know witch one is mounted or not. it will be nice if we can have something in the config file for the tool changer...
Thanks you very much.

**Comment by TypQxQ** [comment]
You can do that with a custom init_gcode configuration. See the documentation. Also check my backup config here:

https://github.com/TypQxQ/DuetBackup/blob/main/qTC-Klipper/config/ktc/TOOLS.cfg

**Reply by ClaudeH68** [reply]
Thanks it's exactly what I'm looking for.  The part that I'm missing was the init_mode = ON_START.

Thank you very much

**Reply by ClaudeH68** [reply]
Hi Andrei, I have a issu when I deselect tools (T_1) and I restart klipper.  I found that when I deselect all tools, the persistant keep the last tool selected, so when I reboot klipper, I got the last tool with the status as ENGAGED so If I select an other tools, I will get le new tools as Active and the last tool (before reboot) as Engaged, so If I do a T_1,  I have the Active tool that excute deselect and also the Engaged tools. so do you know a way to save to the presistence when there is no tools, or not to load the persistence for the status of each tools
<img width="393" alt="Screenshot 2025-01-18 155808" src="https://github.com/user-attachments/assets/af192c88-8b69-4541-b3be-62834692fcb7" />

**Reply by TypQxQ** [reply]
Please check your persistent data file for ktc before and after restart.

**Reply by ClaudeH68** [reply]
After a restrat I got T0 Selected, and I didn't do anything
<img width="370" alt="Screenshot 2025-01-20 070819" src="https://github.com/user-attachments/assets/119412e8-b70a-41cf-9b14-6fa7b72b1b5f" />

**Reply by ClaudeH68** [reply]
if I make a tool dropoff, the persistent is not updated, still show selected_tool : 0

**Comment by TypQxQ** [comment]
You are probably missing a line like this when you disengage your toolhead.:
https://github.com/TypQxQ/DuetBackup/blob/7182d70f029f7d4c872b71a4eb6a4fffedd2f1a4/qTC-Klipper/config/ktc/TOOLS.cfg#L57

**Reply by ClaudeH68** [reply]
Thank Andrei it's exactly what it’s missing.  I have no lock mechanism, so I didn’t fill engage_gcode, and disengage_gcode.

Also do you know a way to change the maximum value of an axis dynamically.
I would like to change the Maximum Y value after I puck up my tools, to remove le tool length

Thanks, you very much 
Claude

**Reply by TypQxQ** [reply]
I fon't think you can. 
I use a optical sensor that I enable when not tool changing and if triggered otherwise it would issue a emergency stop.
Because its optical, the axis can pass by it.
